client.go

```diff
@@ -1507,6 +1507,9 @@ func (c *Client) unsubscribe(channel string, fn func(UnsubscribeResult, error))
 		return
 	}
 	c.sendUnsubscribe(channel, fn)
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+	delete(c.subs, channel)
```

Comment:
Call to `Subscribe` should return Subscription back I suppose – since Centrifuge has a message recovery mechanism and automatic resubscribe to active subscriptions on reconnect.

Reply:
We cannot call `Subscribe` since it's method of Subscription.
To recovery case maybe use `NewSubscription` of `Client` method.
But there is:
`	if _, ok := c.subs[channel]; ok {
		return nil, ErrDuplicateSubscription
	}
`

Reply:
But anyway, with my solution (delete from map) I tried to resolve possible memory leak.

Reply:
I mean that the following use case should continue to work:

```go
subscription.Unsubscribe() // deletes Subscription from Client's map
subscription.Subscribe() // returns Subscription to Client's map
```

This is possible since `Subscription` has a reference to a `Client`. Otherwise once unsubscribed Subscription can not be used again.